### ISSUE TITLE
Blockreassurance requires executable directory

### ISSUE THREAD

**Issue by musicpanda** [PrestaShop/PrestaShop#33808]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When configuring the Blockreassurance module under Windows I get a warning that a directory is not readable or executable:

![reassurance](https://github.com/PrestaShop/PrestaShop/assets/13930107/b500979a-1c53-4165-82a2-45d95d8e7c0e)

When I look for the relevant code I see that this function is called in blockreassurance.php:
`    private function folderUploadFilesHasGoodRights()
    {
        return is_writable($this->folder_file_upload)
            && is_executable($this->folder_file_upload);
    }`

It evades me why a folder that is meant to contain images should be executable. If anything, that looks like a security risk to me.

### Expected behavior

There should be no requirement that this folder is executable.

### Steps to reproduce

Run Prestashop 8.1.0 under Windows
Go into the admin section
Configure the Blockreassurance module.

### PrestaShop version(s) where the bug happened

8.1

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

blockreassurance

**Comment by sallemiines** [comment]
Ping @Hlavtox or @PrestaShop/committers could you please try to reproduce the issue :pray:

**Comment by JBPSA** [comment]
Hi @musicpanda 

is_executable() on Windows will always return false, you can read more [here](https://github.com/PrestaShop/blockreassurance/pull/529).

[PrestaShop works best in Unix-based systems such as Linux and macOS, and while Windows might work, it is not actively supported.](https://devdocs.prestashop-project.org/8/basics/installation/system-requirements/)

**Comment by kpodemski** [comment]
The truth is that we could think about a different way of validating everything inside and not use `exec.

**Comment by boo-code** [comment]
The reported symptom is gone, but the code that removed it introduced a worse defect in the same function, so this should not be closed as fixed.

First, the premise in the report. The execute bit on a *directory* does not mean "run programs in it" - it means "traverse it", and creating a file inside a directory requires both the write and the traverse permission. So checking it is correct on Linux and macOS and is not a security risk. The Windows part of the report was right for a different reason: Windows has no traverse bit, and `is_executable()` there answers about file extensions instead, so it returns false for every directory and the warning appeared on a perfectly usable folder.

That was fixed in `b93b97d` (April 2023), before this issue was opened, with nested conditions that were correct:

```php
if (is_writable($this->folder_file_upload)) {
    if (preg_match('/^[a-zA-Z]{1}\:{1}\\\\{1}/', $this->folder_file_upload) !== 1) {
        return is_executable($this->folder_file_upload);
    }
    return true;
}
return false;
```

Five days later `a554f0d` flattened it into one expression, which is what `dev` carries today:

```php
return is_writable($f) && !(preg_match(...) !== 1) || is_executable($f);
```

`&&` binds tighter than `||`, so that reads `(is_writable && isWindowsPath) || is_executable`. On Linux the left side is always false, which leaves `is_executable()` alone deciding - and a directory only has to be traversable for that to be true. **The writability check, the one thing the function exists for, no longer applies.**

Measured on a real directory, as www-data:

```
path            : /tmp/qa33808   (mode 0555 - readable and traversable, not writable)
is_writable     : false
is_executable   : true
module reports  : folder rights are GOOD
actual upload   : FAILED (permission denied)
```

The configuration page tells the merchant the folder is fine and the upload then fails. Across the four permission combinations the current expression disagrees with reality in one case; the restored version disagrees in none.

Fix prepared against `dev` in the module repository: restore the original meaning, and detect the platform rather than matching the path against a drive-letter pattern, which did not cover UNC paths (`\\server\share`) or relative ones. Verified against the intended pre-flattening logic across both platforms and all four permission combinations - 0 disagreements out of 8 - and against real directories, where it now matches whether a file can actually be created in all four modes.

**Comment by kpodemski** [comment]
@boo-code to avoid problems with CI you can submit PRs during weekends, and maybe after 8PM 👍🏻 this way we can continue doing our part, and the maintenance 

we continuously check the PRa you have sent

**Comment by boo-code** [comment]
@kpodemski Agreed. New pull requests will go out at weekends or after 20:00 CEST from now on.
